EosPayload/drivers/GPS_driver.py

```diff
@@ -41,50 +29,50 @@ def __init__(self, output_directory: str, config: dict) -> None:
     def setup(self) -> None:
         super().setup()
 
-        try:
-            UART
-        except NameError:
-            raise Exception("failed to import UART library")
-
         self.register_thread('device-read', self.device_read)
-
-        UART.setup("UART1")
-        self.uart = serial.Serial(port="/dev/ttyO1", baudrate=9600)
-        self.gps = adafruit_gps.GPS(self.uart, debug=False)
-
-        self.gps.send_command(b"PMTK314,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0")
-        self.gps.send_command(b"PMTK220,1000")
+        self.uart = serial.Serial('/dev/ttyO1', baudrate=9600, timeout=1)
+        self.gps = UbloxGps(self.uart)
 
     def device_read(self, logger: logging.Logger) -> None:
-
+        logger.info("Starting GPS Driver!")
         while True:
-            self.gps.update()
-            if not self.gps.has_fix:
+            geo = self.gps.geo_coords()
+            gps_fix = geo.fixType  # fix type
+            if int(gps_fix) < 3:
                 # Try again if we don't have a fix yet.
                 logger.info("Waiting for fix...")
                 self.thread_sleep(logger, 1)
                 continue
 
-            time_hr = self.gps.timestamp_utc.tm_hour
-            time_min = self.gps.timestamp_utc.tm_min
-            time_sec = self.gps.timestamp_utc.tm_sec
-            time_day = self.gps.timestamp_utc.tm_mday
-            time_month = self.gps.timestamp_utc.tm_mon
-            time_year = self.gps.timestamp_utc.tm_year
+            gps_time = self.gps.date_time()
+            time_sec = gps_time.sec
+            time_min = gps_time.min
+            time_hr = gps_time.hour
+            time_day = gps_time.day
+            time_month = gps_time.month
+            time_year = gps_time.year
+
+            # https://content.u-blox.com/sites/default/files/products/documents/u-blox8-M8_ReceiverDescrProtSpec_UBX-13003221.pdf
+            # page 304
+            gps_lat = geo.lat  # deg
+            gps_long = geo.lon  # deg
+            gps_alt = geo.hMSL  # mm
+            gps_speed = geo.gSpeed  # mm/s (2D speed)
```

Comment:
you need to maintain the units that were being previously used, we have drivers that auto-trigger based on altitude and if you specify mm instead of meters then its going to cause big problems